source/common/common/BUILD

```diff
@@ -414,3 +415,30 @@ envoy_cc_library(
         "@com_google_absl//absl/status:statusor",
     ],
 )
+
+envoy_pch_library(
+    name = "common_pch",
+    external_deps = [
+        "spdlog",
+    ],
+    includes = [
+        "envoy/config/bootstrap/v3/bootstrap.pb.h",
+        "envoy/config/cluster/v3/cluster.pb.h",
+        "envoy/config/core/v3/base.pb.h",
+        "envoy/config/core/v3/config_source.pb.h",
+        "envoy/config/route/v3/route_components.pb.h",
+        "envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.pb.h",
+        "envoy/service/discovery/v3/discovery.pb.h",
+        "spdlog/sinks/android_sink.h",
+        "spdlog/spdlog.h",
+    ],
```

Comment:
Not fan of directly listing those headers here. Can they be extracted from `deps` headers?

Reply:
So the problem is that the `CcInfo`s of our `deps` include _all_ headers transitively that they depend on, so we can't (or at least don't want to) just dump them all in the generated header file we use to create the pch. I tried some of the other attrs like `direct_public_headers` but they all appear to be empty. I'm not sure how those are supposed to fit in with this. One benefit here is that we're quite explicit about what things we want to be pre-compiled vs potentially blowing up this file with all transitive headers